File: app/src/main/java/com/plweegie/android/squash/viewmodels/FaveListViewModelFactory.java
package com.plweegie.android.squash.viewmodels;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.plweegie.android.squash.data.RepoRepository;

public class FaveListViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private final RepoRepository mRepository;

    public FaveListViewModelFactory(RepoRepository repository) {
        this.mRepository = repository;
    }

    @Override
    public <T extends ViewModel> T create(Class<T> modelClass) {
        return (T) new FaveListViewModel(mRepository);
    }
}

File: app/src/main/java/com/plweegie/android/squash/viewmodels/FaveListViewModelFactory.kt
package com.plweegie.android.squash.viewmodels

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

import com.plweegie.android.squash.data.RepoRepository
import javax.inject.Inject
import javax.inject.Provider

class FaveListViewModelFactory @Inject constructor(private val repository: Provider<RepoRepository>) :
        ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel?> create(modelClass: Class<T>): T =
        FaveListViewModel(repository.get()) as T
}
